spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/RewriteViewCommands.scala

```diff
@@ -60,17 +63,23 @@ case class RewriteViewCommands(spark: SparkSession) extends Rule[LogicalPlan] wi
         properties = properties,
         allowExisting = allowExisting,
         replace = replace)
+
+    case a@AlterViewAs(ResolvedV2View(_, ident, _), _, query) =>
```

Comment:
I don't like that this depends on the next case running before this one. I think that I would combine the two.

I think this should be:

```scala
    case a@AlterViewAs(ResolvedView(resolved), _, query) =>
      val q = CTESubstitution.apply(query)
      verifyTemporaryObjectsDontExist(ident, q)
      a.copy(child=resolved, query = q)
```

Reply:
if we want to go this route, then we'd also need to add the same handling for `SetViewProperties(UnresolvedView(...))` / `UnsetViewProperties(UnresolvedView(...))`, hence it seemed better to me to just handle `case u@UnresolvedView(ResolvedView(resolved), _, _, _)` separately.
I'm fine either way, so let me know if you'd still like me to switch to the approach you proposed.